### PR TITLE
DM properties handling  

### PR DESCRIPTION
**Pull request by erandacr**

### CODE REVIEW

File: components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/core/executors/ScriptExecutor.java [hunk 1]
```diff
@@ -86,4 +90,8 @@ private void injectInputVariableToEngine(String inputSchemaName, String inputVar
         scriptEngine.eval("var input" + inputSchemaName.replace(':', '_').replace('=', '_').replace(',', '_') + "="
                 + inputVariable);
     }
+
+    private void injectPropertiesToEngine(String properties) throws ScriptException {
+        scriptEngine.eval("var " + PROPERTIES_OBJECT_NAME + "= " + properties);
```
Comment (virajsenevirathne) [inline components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/core/executors/ScriptExecutor.java]: It is often good to change '=' things to constants. and this line has formatting issue

File: components/mediators/datamapper/org.wso2.carbon.mediator.datamapper/src/main/java/org/wso2/carbon/mediator/datamapper/DataMapperMediator.java [hunk 2]
```diff
@@ -70,6 +84,26 @@
...
+     * @param synCtx Message context
+     * @param key    registry key
+     * @return mapping configuration, inputSchema and outputSchema as inputStreams
+     */
+    private static InputStream getRegistryResource(MessageContext synCtx, String key) {
+        InputStream inputStream = null;
+        Object entry = synCtx.getEntry(key);
+        if (entry instanceof OMTextImpl) {
+            if (log.isDebugEnabled()) {
+                log.debug("Value for the key is ");
```
Comment (virajsenevirathne) [inline components/mediators/datamapper/org.wso2.carbon.mediator.datamapper/src/main/java/org/wso2/carbon/mediator/datamapper/DataMapperMediator.java]: What is the key? value

File: components/mediators/datamapper/org.wso2.carbon.mediator.datamapper/src/main/java/org/wso2/carbon/mediator/datamapper/DataMapperMediator.java [hunk 3]
```diff
@@ -392,23 +434,73 @@ private MappingResource getMappingResource(MessageContext synCtx, String configK
...
+            default:
+                log.warn(contextAndName[0] + " scope is not found. Setting it to an empty value.");
+                value = "";
             }
-            OMTextImpl text = (OMTextImpl) entry;
-            String content = text.getText();
-            inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+            if (value == null) {
+                log.warn(propertyName + "not found. Setting it to an empty value.");
+                value = "";
```
Comment (virajsenevirathne) [inline components/mediators/datamapper/org.wso2.carbon.mediator.datamapper/src/main/java/org/wso2/carbon/mediator/datamapper/DataMapperMediator.java]: "" make this as a constant, you have used it several times in the code.. this is default value or something

File: components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/core/mapper/MappingResource.java [hunk 4]
```diff
@@ -69,6 +75,10 @@ public JSFunction getFunction() {
         return function;
     }
 
+    public List getPropertiesList() {
```
Comment (sumuditha-viraj) [inline components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/core/mapper/MappingResource.java]: It's always better to add a method comment for public methods, even though it is a getter for properties list. We can mentioned which properties etc.

Comment (erandacr) [inline components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/core/mapper/MappingResource.java]: Fixed with https://github.com/wso2/carbon-mediation/pull/748

**Pull request merged**